### PR TITLE
Add `id` to `DEI_Answer` definition

### LINKED ISSUE #12: Adding an id field to DEI Answer
Most of the publishers of 360Giving are spreadsheet publishers, and rely on infrastructure using flatten-tool to publish the data.

flatten-tool is used to auto-generate a template from the schema files. This is available to view [here](https://360-dei.readthedocs.io/en/latest/spreadsheet.html#) for the 360-dei schema.

One-to-many relationships are modelled using separate sheets or via JSON pointer syntax. In the case of DEI_Answers, each grant may have many DEI_Answers. In turn, each DEI_Answer may have many DEI_Classifications, as well as arrays of `DEI_Answer/location/description`.

Without an identifier field on DEI_Answer, this means that one-to-many sheets relating to DEI Answers will need to use the JSON Pointer syntax in addition to the separate sheet or all of the DEI_Classifications will be applied to a single DEI_Answer object represented by the sheet, likely the first. This is because the `id` field in the sheet is specifically related to the grant, and the pointer syntax defaults to `0`.

Unless additional columns are explicitly added by a publisher, with the correct pointer syntax, this will mean that e.g. all DEI Classifications added under `project` for a grant, will relate to the first DEI Answer for that project.

This may be fine for some publisher and cases where there is only a few answers but will result in potentially unwieldy spreadsheets for cases where multiple options are selected across different grants.

This affects, at least, the following fields wherever DEI_Answer occurs:

* `dei_available_options` (array of strings from the askedStatus codelist)
* `dei_classification_available_options` (array of `DEI_Classification`)
* `classification` (array of `DEI_Classification`)
* `location` (array of objects with `location/description`)

Adding an identifier field to DEI_Answer will allow e.g. the "pro_dei_classification_available" sheet to support each classification available on a single row, identifiable by the DEI Answer identifier rather than the grant identifier. This means that it'll be easier to input and manage this data for spreadsheet users.

### PR DESCRIPTION
**Pull request by mrshll1001**
Added a required id field to DEI_Answer, which can either be a string or an integer

Resolves #12

### CODE REVIEW

**Review by KDuerden — APPROVED** [review]
this looks good to me, thank you.

File: schema/360-giving-schema-extension.json [hunk 1]
```diff
@@ -40,7 +40,17 @@
...
+	  "id"
+	],
         "properties": {
+	  "id": {
+	    "title": "DEI Answer Identifier",
+	    "type": [
+	      "string",
+	      "integer"
+	    ]
+	  },
```
Comment (Bjwebb) [inline schema/360-giving-schema-extension.json]: Looks like the indentation is wrong.

Comment (mrshll1001) [reply schema/360-giving-schema-extension.json]: Resolved now.

**Comment by Bjwebb** [comment]
I've tested this runs through CoVE correctly, by populating the template with a bit of example data: [360-giving-dei-bens-example.xlsx](https://github.com/ThreeSixtyGiving/360-dei/files/11147561/360-giving-dei-bens-example.xlsx)

**Review by Bjwebb — APPROVED** [review]

**Pull request merged**